external_deps/build.sh

```diff
@@ -59,54 +59,65 @@ LDFLAGS=''
 # Extract an archive into the given subdirectory of the build dir and cd to it
 # Usage: extract <filename> <directory>
 extract() {
-	rm -rf "${BUILD_DIR}/${2}"
-	mkdir -p "${BUILD_DIR}/${2}"
+	rm -rf "${2}"
+	mkdir -p "${2}"
 	case "${1}" in
 	*.tar.bz2)
-		tar xjf "${DOWNLOAD_DIR}/${1}" -C "${BUILD_DIR}/${2}"
+		tar xjf "${1}" -C "${2}"
 		;;
 	*.tar.xz)
-		tar xJf "${DOWNLOAD_DIR}/${1}" -C "${BUILD_DIR}/${2}"
+		tar xJf "${1}" -C "${2}"
 		;;
 	*.tar.gz|*.tgz)
-		tar xzf "${DOWNLOAD_DIR}/${1}" -C "${BUILD_DIR}/${2}"
+		tar xzf "${1}" -C "${2}"
 		;;
 	*.zip)
-		unzip -d "${BUILD_DIR}/${2}" "${DOWNLOAD_DIR}/${1}"
+		unzip -d "${2}" "${1}"
 		;;
 	*.cygtar.bz2)
 		# Some Windows NaCl SDK packages have incorrect symlinks, so use
 		# cygtar to extract them.
-		"${SCRIPT_DIR}/cygtar.py" -xjf "${DOWNLOAD_DIR}/${1}" -C "${BUILD_DIR}/${2}"
+		"${SCRIPT_DIR}/cygtar.py" -xjf "${1}" -C "${2}"
 		;;
 	*.dmg)
-		mkdir -p "${BUILD_DIR}/${2}-dmg"
-		hdiutil attach -mountpoint "${BUILD_DIR}/${2}-dmg" "${DOWNLOAD_DIR}/${1}"
-		cp -R "${BUILD_DIR}/${2}-dmg/"* "${BUILD_DIR}/${2}/"
-		hdiutil detach "${BUILD_DIR}/${2}-dmg"
-		rmdir "${BUILD_DIR}/${2}-dmg"
+		mkdir -p "${2}-dmg"
+		hdiutil attach -mountpoint "${2}-dmg" "${1}"
+		cp -R "${2}-dmg/"* "${2}/"
+		hdiutil detach "${2}-dmg"
+		rmdir "${2}-dmg"
 		;;
 	*)
 		echo "Unknown archive type for ${1}"
 		exit 1
 		;;
 	esac
-	cd "${BUILD_DIR}/${2}"
+	cd "${2}"
 }
 
 # Download a file if it doesn't exist yet, and extract it into the build dir
 # Usage: download <filename> <URL> <dir>
 download() {
-	if [ ! -f "${DOWNLOAD_DIR}/${1}" ]; then
-		"${CURL}" -L --fail -o "${DOWNLOAD_DIR}/${1}" "${2}"
-	fi
-	extract "${1}" "${3}"
+	local extract_dir="${BUILD_DIR}/${1}"; shift
+	local tarball_file="${DOWNLOAD_DIR}/${1}"; shift
+	while [ ! -f "${tarball_file}" ]; do
+		[ -n "${1:-}" ]
```

Comment:
Should we do something like:
`[ -n "${1:-}" ] ||( echo "No more URLs passed in!" >&2 && exit -1)`